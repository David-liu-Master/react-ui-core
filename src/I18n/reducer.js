import produce from 'immer';
import { CHANGE_LANGUAGE } from './constants';

const initialState = {
  language: 'en'
};

export default (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case CHANGE_LANGUAGE:
        draft.language = action.payload.language;
    }
  });
