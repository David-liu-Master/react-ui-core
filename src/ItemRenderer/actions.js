import { CHANGE_NOTE, CHANGE_RESPONSE } from './constants';

export const changeNote = note => ({
  type: CHANGE_NOTE,
  payload: {
    note
  }
});

export const changeResponse = (response, id) => ({
  type: CHANGE_RESPONSE,
  payload: {
    response,
    id
  }
});
