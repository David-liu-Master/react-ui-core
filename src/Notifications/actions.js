import {
  ADD_NOTIFICATION,
  REMOVE_NOTIFICATION,
  CLEAR_NOTIFICATIONS
} from './constants';

export function addNotification(notification) {
  return {
    type: ADD_NOTIFICATION,
    payload: {
      notification
    }
  };
}

export function removeNotification() {
  return {
    type: REMOVE_NOTIFICATION
  };
}

export function clearNotifications() {
  return {
    type: CLEAR_NOTIFICATIONS
  };
}
