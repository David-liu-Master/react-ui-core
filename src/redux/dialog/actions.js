import { INIT, OPEN, CANCEL, CONFIRM } from './constants';

export function init(dialogId) {
  return {
    type: INIT,
    payload: {
      dialogId
    }
  };
}

export function open(dialogId) {
  return {
    type: OPEN,
    payload: {
      dialogId
    }
  };
}

export function confirm(dialogId) {
  return {
    type: CONFIRM,
    payload: {
      dialogId
    }
  };
}

export function cancel(dialogId) {
  return {
    type: CANCEL,
    payload: {
      dialogId
    }
  };
}
