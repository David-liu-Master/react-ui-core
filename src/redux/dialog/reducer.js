import produce from 'immer';

import { INIT, OPEN, CONFIRM, CANCEL } from './constants';

const initialState = {};

export default (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case INIT:
        if (!(action.payload.dialogId in draft)) {
          draft[action.payload.dialogId] = false;
        }
        break;
      case OPEN:
        draft[action.payload.dialogId] = true;
        break;
      case CANCEL:
      case CONFIRM:
        draft[action.payload.dialogId] = false;
        break;
    }
  });
