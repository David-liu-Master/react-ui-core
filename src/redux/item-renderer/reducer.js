import produce from 'immer';

import { CHANGE_NOTE, CHANGE_RESPONSE } from './constants';

const initalState = { responses: {}, note: '' };

export default (state = initalState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case CHANGE_NOTE:
        draft.note = action.payload.note;
        break;
      case CHANGE_RESPONSE:
        draft.responses[action.payload.id] = action.payload.response;
        break;
    }
  });
