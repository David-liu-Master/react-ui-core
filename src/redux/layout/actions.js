import {
  TOGGLE_DRAWER,
  REMOTE_SECTION_REQUEST,
  REMOTE_SECTION_ROLLBACK,
  REMOTE_SECTION_COMMIT
} from './constants';

export function toggleDrawer() {
  return {
    type: TOGGLE_DRAWER
  };
}

export function fetchRemoteSection(url) {
  return dispatch => {
    dispatch(fetchRemoteSectionRequest());
    fetch(url, {
      method: 'GET',
      cors: 'no-cors'
    })
      .then(function(response) {
        return response.json();
      })
      .then(function(json) {
        return dispatch(fetchRemoteSectionCommit(url, json));
      })
      .catch(function(err) {
        return dispatch(fetchRemoteSectionRollback());
      });
  };
}

function fetchRemoteSectionRequest() {
  return {
    type: REMOTE_SECTION_REQUEST
  };
}

function fetchRemoteSectionCommit(url, sections) {
  return {
    type: REMOTE_SECTION_COMMIT,
    payload: {
      url,
      sections
    }
  };
}

function fetchRemoteSectionRollback() {
  return {
    type: REMOTE_SECTION_ROLLBACK
  };
}
