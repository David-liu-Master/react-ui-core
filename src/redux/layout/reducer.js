import produce from 'immer';

import { TOGGLE_DRAWER } from './constants';

const initialState = {
  drawerOpen: false
};

export default (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case TOGGLE_DRAWER:
        draft.drawerOpen = !draft.drawerOpen;
    }
  });
