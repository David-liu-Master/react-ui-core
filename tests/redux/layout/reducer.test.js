import reducer from '../../../src/redux/layout/reducer';
import { toggleDrawer } from '../../../src/redux/layout/actions';

const mockState = {
  drawerOpen: true
};

describe('layout reducer', () => {
  it('should return initialState', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });
  it('should reduce toggleDrawer', () => {
    expect(reducer(undefined, toggleDrawer())).toMatchSnapshot();
    expect(reducer(mockState, toggleDrawer())).toEqual({
      drawerOpen: false
    });
  });
});
