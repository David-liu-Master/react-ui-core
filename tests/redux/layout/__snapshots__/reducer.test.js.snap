// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layout reducer should reduce toggleDrawer 1`] = `
Object {
  "drawerOpen": true,
  "isRequesting": false,
  "remoteSections": Object {},
  "requestingFailed": false,
}
`;

exports[`layout reducer should return initialState 1`] = `
Object {
  "drawerOpen": false,
  "isRequesting": false,
  "remoteSections": Object {},
  "requestingFailed": false,
}
`;
